File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        result[0] = 1;
        result[n - 1] = 1;

        int prefixProduct = 1;
        for (int i = 1; i < n; i++) {
            prefixProduct *= nums[i - 1];
            result[i] = prefixProduct;
        }

        int suffixProduct = 1;
        for (int i = n - 2; i >= 0; i--) {
            suffixProduct *= nums[i + 1];
            result[i] *= suffixProduct;
        }
        
        return result;
    }
}  

File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        dict = {}
        for num in nums:
            if num in dict:
                dict[num] += 1
            else:
                dict[num] = 1
        
        for key, value in dict.items():
            if value > (len(nums) // 2):
                return key